### ISSUE TITLE
Accessing the VAD of kyutai/stt-1b-en_fr-candle through pytorch

### ISSUE THREAD

**Issue by Akashnandi1** [kyutai-labs/delayed-streams-modeling#86]
I found the below notes in the [stt_from_mic_rust_server.py](https://github.com/kyutai-labs/delayed-streams-modeling/blob/main/scripts/stt_from_mic_rust_server.py)

```
# The VAD has several prediction heads, each of which tries to determine whether there
# has been a pause of a given length. The lengths are 0.5, 1.0, 2.0, and 3.0 seconds.
# Lower indices predict pauses more aggressively. In Unmute, we use 2.0 seconds = index 2.
```

Based on the above text, my understanding is that the VAD head should have 4 logits.

But when I see the description of the LM Model, I see the below description. Here, I assume extra heads refers to the VAD. There are 4 of them, which also makes sense to me. However, why is the output_dim of each of these heads 6? 

Request you to kindly clarify how to use the VAD heads when using pytorch.

Thanks!

`LMModel(

  (emb): ModuleList(
    (0-31): 32 x ScaledEmbedding(2049, 2048)
  )
  (text_emb): ScaledEmbedding(8001, 2048)
  (text_linear): Linear(in_features=2048, out_features=8000, bias=False)
  (transformer): StreamingTransformer(
    (rope): RotaryEmbedding()
    (layers): ModuleList(
      (0-15): 16 x StreamingTransformerLayer(
        (self_attn): StreamingMultiheadAttention(
          (rope): RotaryEmbedding()
          (out_projs): ModuleList(
            (0): Linear(in_features=2048, out_features=2048, bias=False)
          )
          (in_projs): ModuleList(
            (0): Linear(in_features=2048, out_features=6144, bias=False)
          )
        )
        (norm1): RMSNorm()
        (norm2): RMSNorm()
        (gating): ActivationGating(
          (linear_in): Linear(in_features=2048, out_features=11264, bias=False)
          (linear_out): Linear(in_features=5632, out_features=2048, bias=False)
        )
        (layer_scale_1): Identity()
        (layer_scale_2): Identity()
      )
    )
  )
  (out_norm): RMSNorm()
  (depformer_in): ModuleList()
  (extra_heads): ModuleList(
    (0-3): 4 x Linear(in_features=2048, out_features=6, bias=False)
  )
  (linears): ModuleList()
  (condition_provider): ConditionProvider(
    (conditioners): ModuleDict()
  )
)`

**Comment by LaurentMazare** [comment]
The 4 vad extra-heads are for different time horizons as you noticed, and for each head we consider the probability to have 0, 1, 2, 3, 4, 5 words left, hence the 6 logits, see this [comment](https://github.com/kyutai-labs/delayed-streams-modeling/issues/22#issuecomment-3048114414) for a bit more details.
